components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDao.kt

```diff
@@ -23,6 +24,9 @@ internal interface RecentAppsDao {
     @Update
     fun updateRecentApp(recentApp: RecentAppEntity)
 
+    @Delete
+    fun deleteRecentApp(packageName: RecentAppEntity)
```

Comment:
The parameter `packageName` should be renamed to `activityName`as now it is the[ primary key](https://github.com/mozilla-mobile/android-components/blob/311864c26b1b3c779936701cbcb3ca4a99dff014/components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppEntity.kt#L19).

Also we can take advantage that `activityName` is primary key to improve the delete statement.
Instead of passing the object, we could pass the primary key (`activityName`), this way we can avoid having to query the table before and then deleting. You can find an example [here](https://github.com/mozilla-mobile/android-components/blob/311864c26b1b3c779936701cbcb3ca4a99dff014/components/feature/addons/src/main/java/mozilla/components/feature/addons/update/db/UpdateAttemptDao.kt#L23)